cmake/SetupCxxFlags.cmake

```diff
@@ -81,6 +81,19 @@ set_property(TARGET SpectreFlags
   $<$<COMPILE_LANGUAGE:CXX>:-ffp-contract=on>
   $<$<COMPILE_LANGUAGE:Fortran>:-ffp-contract=on>)
 
+# Make std::fma be evaluated as an FMA under the hood, always.
+#
+# Define for C as well since C99 has an fma function.
+set_property(TARGET SpectreFlags
+  APPEND PROPERTY
+  INTERFACE_COMPILE_DEFINITIONS
+  $<$<COMPILE_LANGUAGE:CXX>:FP_FAST_FMA>
+  $<$<COMPILE_LANGUAGE:C>:FP_FAST_FMA>
+  $<$<COMPILE_LANGUAGE:CXX>:FP_FAST_FMAF>
+  $<$<COMPILE_LANGUAGE:C>:FP_FAST_FMAF>
+  $<$<COMPILE_LANGUAGE:CXX>:FP_FAST_FMAL>
+  $<$<COMPILE_LANGUAGE:C>:FP_FAST_FMAL>)
```

Comment:
I think these macros are output from the standard library reporting support for features, not input to the library requesting features.  They should be set by `-mfma` (probably included in `-march` if appropriate).  Testing with gcc on my machine doesn't show any effect from defining these, but only gives an fma instruction with `-mfma` (which also defines the macros).

Reply:
In my tests GCC defined them but clang unfortunately didn't, even if I have `-O3 -mfma -march=native` on a machine that supports fma. I found this rather surprising and annoying...

Reply:
You are correct that clang is missing the definitions, but defining them on the command line still doesn't enable fma instructions or anything.

Reply:
Ah, I see what you're saying now. I misinterpreted the cppreference wording as "You can define these to enable FMA" instead of "If these are defined, you are guaranteed an FMA". I'll drop the commit. Thanks!